iree/turbine/kernel/wave/thread_shape_analysis.py

```diff
@@ -155,11 +161,26 @@ def determine_thread_shapes(trace: CapturedTrace):
     for anchor_op in anchor_ops:
         custom = get_custom(anchor_op)
         index_sizes = get_custom_dim_sizes(custom)
-        if isinstance(custom, (Read, ReduceOp)):
+        if isinstance(custom, (Read, GetResult)):
             fwd_slice = capture_forward_slice(custom.fx_node, propagatable_op)
             thread_size_to_ops[index_sizes] = thread_size_to_ops.get(
                 index_sizes, set([])
             ).union(fwd_slice)
+        elif isinstance(custom, ReduceOp):
+            fwd_slice = capture_forward_slice(custom.fx_node, propagatable_op)
+            bwd_slice = set()
+            if custom.init != None and not isinstance(
```

Comment:
similar reason for in chained matmul, why we do not expand on acc if the acc is also a matmul

Reply:
we can imagine it's a reduction into another reduction, and we use the 1st reduction as an init to the second. It may try to slice/attach information all the operands of the 1st reduction, which is no good. :)

Reply:
right so if you didn't have this check, then you would do a bwd_slice on the first reduction, which would overwrite the information from the first reduction.

Reply:
yeap!